### PR TITLE
feat(web): 🎸 dotLottie state machines integration

### PR DESCRIPTION
**Pull request by theashraf**
Changes: 
- upgraded dotlottie-rs v0.1.23 wasm bindings
- added startStateMachine method 
- added stopStateMachine method
- added postStateMachineEvent method 
- setup/cleanup dotLottie state machine listeners
- dotlottie-web-example update with state machines

### CODE REVIEW

File: .changeset/gorgeous-seahorses-look.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+dotLottie.postStateMachineEvent("OnPointerUp: 0.0 0.0");   // Post a pointer up event
+dotLottie.postStateMachineEvent("OnPointerMove: 0.0 0.0"); // Post a pointer move event
+dotLottie.postStateMachineEvent("OnPointerEnter: 0.0 0.0"); // Post a pointer enter event
+dotLottie.postStateMachineEvent("OnPointerExit: 0.0 0.0");  // Post a pointer exit event
+dotLottie.postStateMachineEvent("OnComplete");             // Post a complete event
+
+ const smStopped = dotLottie.stopStateMachine(); // End the started state machine
+```
+
+> `DotLottie` will automatically set up and clean up the pointer events on the provided canvas and other animation playback events for you when you call `startStateMachine` and `stopStateMachine`.
```
Comment (samuelOsborne) [inline .changeset/gorgeous-seahorses-look.md:43]: @theashraf Regarding this point, this is implemented right? I can't see a call to `post_serialized_event`

Comment (theashraf) [reply .changeset/gorgeous-seahorses-look.md:43]: It's called postEventPayload in the Emscripten bindings

https://github.com/LottieFiles/dotlottie-rs/blob/main/dotlottie-ffi/emscripten_bindings.cpp#L158

Comment (theashraf) [reply .changeset/gorgeous-seahorses-look.md:43]: I think we should give it a better name to indicate that it posts an event to the state machine engine, that's why I created a wrapper for postEventPayload and renamed it to postStateMachineEvent

Comment (samuelOsborne) [reply .changeset/gorgeous-seahorses-look.md:43]: sorry @theashraf  i wrote the wrong method, I meant this one `state_machine_framework_setup`. 

This returns the listeners to setup but I dont see it called in the code and in that changeset you say the listeners are auto setup

**Review by samuelOsborne — APPROVED** [review]

**Pull request merged**